routers/api/v1/repo/file.go

```diff
@@ -45,3 +45,27 @@ func GetArchive(ctx *context.APIContext) {
 
 	repo.Download(ctx.Context)
 }
+
+func GetEditorconfig(ctx *context.APIContext) {
+	if !ctx.Repo.HasAccess() {
```

Comment:
Access is been checked in [`repoAssignment`](https://github.com/gogits/gogs/blob/3738b6399eb360fef3f987e8cb9d0121f080ed61/routers/api/v1/api.go#L76) already.